File: backend/Services/Parsers/BandleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class BandleParser : BasicScoreResultParser
    {
        private ILogger<BandleParser> _logger;

        public BandleParser(ILogger<BandleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override bool Default => false;
        public override string GameName => "Bandle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{GameName} .+?(?<{ScoreGroup}>[\\d|x])/\\d");
        protected override string? ExtraContent => Url;
        public override string Url => "https://bandle.app/";
    }
}

File: lambdas/src/parsers/bandle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class BandleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Bandle';
  readonly helpText = null;
  readonly url = 'https://bandle.app/';
  protected readonly extraContent = this.url;
  
  // Matches "Bandle #611 x/6" or "Bandle #595 1/6"
  // Note: C# uses [\d|x] which incorrectly includes literal pipe - fixed here to [\dx]
  protected readonly parser = new RegExp(
    `${this.gameName} .+?(?<${BasicScoreResultParser.SCORE_GROUP}>[\\dx])/\\d`
  );
}
